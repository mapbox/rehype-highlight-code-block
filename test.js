'use strict';

const rehype = require('rehype');
const dedent = require('dedent');
const rehypeHighlightCodeBlock = require('.');

const processHtml = html => {
  return rehype()
    .data('settings', { fragment: true })
    .use(rehypeHighlightCodeBlock, {
      highlight: jest.fn(() => '<span>mockSyntaxHighlighting</span>')
    })
    .processSync(html)
    .toString();
};

describe('rehypeHighlightCodeBlock', () => {
  test('mocks syntax highlighting properly', () => {
    const result = processHtml(dedent`
      <div>
        <p>foo</p>
        <pre><code>p { color: red }</code></pre>
        <p>bar</p>
      </div>
    `);
    expect(result).toMatchSnapshot();
  });
  test('does nothing to code block, missing <pre></pre>', () => {
    const result = processHtml(dedent`
      <code>p { color: red }</code>
    `);
    expect(result).toMatchSnapshot();
  });
  test('does nothing to code block, missing <code></code>', () => {
    const result = processHtml(dedent`
      <pre>p { color: red }</pre>
    `);
    expect(result).toMatchSnapshot();
  });
  test('does nothing to code block, missing pre > code', () => {
    const result = processHtml(dedent`
      <div>p { color: red }</div>
    `);
    expect(result).toMatchSnapshot();
  });
});
