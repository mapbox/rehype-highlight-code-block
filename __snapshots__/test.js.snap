// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rehypeHighlightCodeBlock does nothing to code block, missing <code></code> 1`] = `"<pre>p { color: red }</pre>"`;

exports[`rehypeHighlightCodeBlock does nothing to code block, missing <pre></pre> 1`] = `"<code>p { color: red }</code>"`;

exports[`rehypeHighlightCodeBlock does nothing to code block, missing pre > code 1`] = `"<div>p { color: red }</div>"`;

exports[`rehypeHighlightCodeBlock mocks syntax highlighting properly 1`] = `
"<div>
  <p>foo</p>
  <pre><code><span>mockSyntaxHighlighting</span></code></pre>
  <p>bar</p>
</div>"
`;
